homeassistant/components/blue_current/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for Blue Current integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from bluecurrent_api import Client
+from bluecurrent_api.exceptions import (
+    AlreadyConnected,
+    InvalidApiToken,
+    RequestLimitReached,
+    WebsocketError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, LOGGER
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_API_TOKEN): str})
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Blue Current."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        client = Client()
+        errors = {}
+        if user_input is not None:
+            api_token = user_input[CONF_API_TOKEN]
+            self._async_abort_entries_match(user_input)
+
+            try:
+                await client.validate_api_token(api_token)
+                email = await client.get_email()
+            except WebsocketError:
+                errors["base"] = "cannot_connect"
+            except RequestLimitReached:
+                errors["base"] = "limit_reached"
+            except AlreadyConnected:
+                errors["base"] = "already_connected"
+            except InvalidApiToken:
+                errors["base"] = "invalid_token"
+            except Exception:  # pylint: disable=broad-except
+                LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            else:
+                self._abort_if_unique_id_configured()
```

Comment:
Set email as the unique_id here, (or something else unique that we can get). In your last changes you actually removed setting the unique id, so what are we checking here?

By using a unique useridentifier (as email) here, someone can't add 2 API keys (probably isn't possible, but imagine that one API key expired and someone enters a new one. With the current solution it would create a second entry, with my proposal it wouldn't work since the email we got back is already used in a config entry)

Reply:
Removed the `_abort_if_unique_id_configured()` so that in the event a new token is generated it it possible to create a new entry. 

I kept the `_async_abort_entries_match()` since that will always return an error because the websocket server only allows one connection and its faster to block it before connecting to the websocket.

Reply:
In the event of a new token, don't we want to keep using the old entry?

Reply:
Yes, that is what the reauth stuff did. I have to do that in another pr right?

Reply:
Yes yes, but what I try to explain is:
1. someones token expired
2. they get a new token
3. they can setup the integration again
4. they now have 1 working 1 broken integration.

Is this possible now? I think using email as unique_id would be a good solution to counter this. Is there something else available? Now I am typing this I am thinking someone might change email. What are your thoughts on this?

Reply:
Yes, the flow is like that.

Could you explain what you want to counter?

Before pushing i did make the following to update the token if an entry exists but thought this should be in the reauth pr.
```python
    else:
        entry = await self.async_set_unique_id(email)

        if entry:
            self.hass.config_entries.async_update_entry(entry, data=user_input)
            await self.hass.config_entries.async_reload(entry.entry_id)

            return self.async_abort(reason="updated_token")

        return self.async_create_entry(title=email, data=user_input)
```


There are no constant values, I could use the account name, but I think email is the better option.

Reply:
OOh right you just told about 2 accounts accessing at the same time. Let me sleep about this for a night.